### PR TITLE
Rename components and fix config

### PR DESCRIPTION
**Pull request by Darktex**
# TLDR
Renames the following:

- All Actors are now Workers
- PyTorchActor --> TrainingWorker
- RefActor --> PostprocessingWorker
- DataCollector is left untouched

Furthermore, I have restructured the config to make it clear what parts belong to who:
- We now have a `inference:`, a `training:`, a `postprocessing:` and an `orchestration:` section in the config which maps to our components

Ran successfully: https://wandb.ai/texx0/torchtune/runs/lez2c2v6?nw=nwusertexx0

### CODE REVIEW

**Review by mikaylagawarecki — APPROVED** [review]

File: recipes/configs/dev/qwen3B_async_grpo.yaml [hunk 1]
```diff
@@ -63,64 +62,65 @@ ref_checkpointer:
...
-vllm:
-  num_workers: 4
+inference:
+  engine: vllm
+  top_k: null
+  temperature: 1.0
   tp_size: 1
+  max_generated_tokens: 512
   batch_size: 1
+  group_size: 16
```
[the remark this answers is not in the document]
Comment (Darktex) [reply recipes/configs/dev/qwen3B_async_grpo.yaml:72]: Hmm, but you can use monte-carlo advantage estimation on many more algos, right? For example, we are seeing people trying simple REINFORCE with monte-carlo advantage estimation (https://arxiv.org/abs/2402.14740) so this is gonna be a feature of many algos, right?
